homeassistant/components/unifi/device_tracker.py

```diff
@@ -453,13 +308,36 @@ def async_update_state(self, event: ItemEvent, obj_id: str) -> None:
                 + description.heartbeat_timedelta_fn(self.controller, self._obj_id),
             )
 
+    @callback
+    def async_event_callback(self, event: Event) -> None:
+        """Event subscription callback."""
+        if event.mac != self._obj_id or self._ignore_events:
+            return
+
+        description = self.entity_description
+        assert isinstance(description.event_to_subscribe, tuple)
+        assert isinstance(description.event_is_on, tuple)
+
+        if event.key in description.event_is_on:
+            self.controller.async_heartbeat(self.unique_id)
+            self._is_connected = True
+            self.async_write_ha_state()
+        else:
```

Comment:
Think this line could be replaced with `return` and skip the `else`

Reply:
I agree!